### PR TITLE
DeleteQuery shouldn't implicitly EndQuery.

### PR DESCRIPTION
**Pull request by jdashg**
Previous tests and implementations implicitly EndQuery if DeleteQuery is
called on an active query.

### CODE REVIEW

**Comment by jdarpinian** [comment]
Hmm. This change seems reasonable to me, for consistency with other parts of the API. But I want to make sure others agree before merging.

**Comment by jdashg** [comment]
Yeah, we should discuss spec changes on conference calls. No rush for this.

File: sdk/tests/conformance2/query/query.html [hunk 1]
```diff
@@ -107,17 +107,43 @@
...
     gl.deleteQuery(q1);
     gl.deleteQuery(q2);
     wtu.glErrorShouldBe(gl, gl.NO_ERROR, "deleting queries should not produce errors");
 
+    shouldBeNonNull("gl.getQuery(gl.ANY_SAMPLES_PASSED, gl.CURRENT_QUERY)");
+    shouldBeNull("gl.getQuery(gl.ANY_SAMPLES_PASSED_CONSERVATIVE, gl.CURRENT_QUERY)");
+    shouldBeNonNull("gl.getQuery(gl.TRANSFORM_FEEDBACK_PRIMITIVES_WRITTEN, gl.CURRENT_QUERY)");
+
+    gl.endQuery(gl.TRANSFORM_FEEDBACK_PRIMITIVES_WRITTEN);
+    wtu.glErrorShouldBe(gl, 0, "Ending a delete query is fine.");
```
Comment (kenrussell) [inline sdk/tests/conformance2/query/query.html]: typo: delete -> deleted

File: sdk/tests/conformance2/query/query.html [hunk 2]
```diff
@@ -107,17 +107,43 @@
...
+    shouldBeNonNull("gl.getQuery(gl.ANY_SAMPLES_PASSED, gl.CURRENT_QUERY)");
+    shouldBeNull("gl.getQuery(gl.ANY_SAMPLES_PASSED_CONSERVATIVE, gl.CURRENT_QUERY)");
     shouldBeNull("gl.getQuery(gl.TRANSFORM_FEEDBACK_PRIMITIVES_WRITTEN, gl.CURRENT_QUERY)");
+
+    gl.endQuery(gl.ANY_SAMPLES_PASSED_CONSERVATIVE);
+    wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION,
+            "Can't endQuery ANY_SAMPLES_PASSED via ANY_SAMPLES_PASSED_CONSERVATIVE.");
+
+    gl.endQuery(gl.ANY_SAMPLES_PASSED);
+    wtu.glErrorShouldBe(gl, 0, "Ending a delete query is fine.");
```
Comment (kenrussell) [inline sdk/tests/conformance2/query/query.html]: typo: delete -> deleted

**Comment by jdashg** [comment]
Superseded by #2991.

**Pull request closed without merging**